### PR TITLE
E2E: Refactor Elasticsearch test to use [Skipped] Tags properly.

### PR DESCRIPTION
**Pull request by jayunit100**
WARNING: THIS PR IS UNTESTED !!!!!

We don't want to inflate the e2e test results.... so this makes the E2E elasticsearch tests so that they use the _It_ syntax, so that they are Skipped rather than returning _passed_.

Hope the _elasticsearch folks can give it a look, or just add a commit on top of this to fix_ if I've made any mistakes.  I think this should work just fine, though.   

I dont run tests on GCE so I cant confirm it works perfectly.

### CODE REVIEW

File: test/e2e/es_cluster_logging.go [hunk 1]
```diff
@@ -41,9 +41,9 @@ var _ = Describe("Cluster level logging using Elasticsearch", func() {
 		Expect(err).NotTo(HaveOccurred())
 	})
 
-	It("should check that logs from pods on all nodes are ingested into Elasticsearch", func() {
-		ClusterLevelLoggingWithElasticsearch(c)
-	})
+	//It("should check that logs from pods on all nodes are ingested into Elasticsearch", func() {
+	ClusterLevelLoggingWithElasticsearch(c)
```
Comment (jayunit100) [inline test/e2e/es_cluster_logging.go]: we can remove these comments after the review when rebasing.

**Comment by jayunit100** [comment]
@satnam6502 Looks like you worked on this test, so maybe you will be best to review this.

**Comment by satnam6502** [comment]
Why skip this test? There is nothing cloud specific about it and it should run on all cloud providers.

**Comment by jayunit100** [comment]
There is a boolean flag in there which makes it skip unless GCE.  I have no idea what its there for.  I just made the skipping so that its done properly.

**Comment by satnam6502** [comment]
OK. Let me take a look.

**Comment by satnam6502** [comment]
I will test this PR on a GCE instance configured with Cloud Logging to ensure it is skipped.

**Comment by satnam6502** [comment]
**Problem.** This seems to skip the test when it should run. Output from a cluster created with Fluentd to Elasticsearch logging. Please investigate.
```
$ go run hack/e2e.go --v --test --test_args="--ginkgo.focus=.*Cluster.*"
2015/05/11 11:18:27 Running: get status
Project: kubernetes-satnam
Zone: us-central1-b
Client Version: version.Info{Major:"0", Minor:"16+", GitVersion:"v0.16.2-710-g2709bb3987698a-dirty", GitCommit:"2709bb3987698ab63af5d8cc60120fe7cb0eda52", GitTreeState:"dirty"}
Server Version: version.Info{Major:"0", Minor:"16+", GitVersion:"v0.16.2-710-g2709bb3987698a-dirty", GitCommit:"2709bb3987698ab63af5d8cc60120fe7cb0eda52", GitTreeState:"dirty"}
Waiting at most 1200 seconds for all pods to be 'Running'
Found 14 pods with statuses: Running Running Running Running Running Running Running Running Running Running Running Running Running Running
All pods are 'Running'
Project: kubernetes-satnam
Zone: us-central1-b
2015/05/11 11:18:31 Running: Ginkgo tests
Setting up for KUBERNETES_PROVIDER="gce".
Project: kubernetes-satnam
Zone: us-central1-b
Running Suite: Kubernetes e2e Suite run 1 of 1
==============================================
Random Seed: 1431368313 - Will randomize all specs
Will run 0 of 45 specs

SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
Ran 0 of 45 Specs in 0.000 seconds
SUCCESS! -- 0 Passed | 0 Failed | 0 Pending | 45 Skipped I0511 11:18:33.507621   16475 driver.go:96] All tests pass
satnam@satnam-linux:~/gocode/src/github.com/GoogleCloudPlatform/kubernetes$ kubectl get pods
POD                                            IP           CONTAINER(S)              IMAGE(S)                                                                            HOST                                    LABELS                                                          STATUS    CREATED          MESSAGE
elasticsearch-logging-ky96p                    10.244.0.2                                                                                                                 kubernetes-minion-89r2/104.197.69.110   kubernetes.io/cluster-service=true,name=elasticsearch-logging   Running   2 days           
                                                            elasticsearch-logging     gcr.io/google_containers/elasticsearch:1.0                                                                                                                                                  Running   2 days           
etcd-server-kubernetes-master                                                                                                                                             kubernetes-master/                      <none>                                                          Running   2 days           
                                                            etcd-container            gcr.io/google_containers/etcd:2.0.9                                                                                                                                                         Running   17 hours         last termination: exit code 0
fluentd-elasticsearch-kubernetes-minion-89r2   10.244.0.4                                                                                                                 kubernetes-minion-89r2/104.197.69.110   <none>                                                          Running   2 days           
                                                            fluentd-elasticsearch     gcr.io/google_containers/fluentd-elasticsearch:1.5                                                                                                                                          Running   2 days           
fluentd-elasticsearch-kubernetes-minion-kxcb   10.244.1.4                                                                                                                 kubernetes-minion-kxcb/104.197.16.133   <none>                                                          Running   2 days           
                                                            fluentd-elasticsearch     gcr.io/google_containers/fluentd-elasticsearch:1.5                                                                                                                                          Running   2 days           
fluentd-elasticsearch-kubernetes-minion-t8m7   10.244.2.2                                                                                                                 kubernetes-minion-t8m7/104.197.77.206   <none>                                                          Running   2 days           
                                                            fluentd-elasticsearch     gcr.io/google_containers/fluentd-elasticsearch:1.5                                                                                                                                          Running   2 days           
fluentd-elasticsearch-kubernetes-minion-xf0z   10.244.3.2                                                                                                                 kubernetes-minion-xf0z/104.197.12.87    <none>                                                          Running   2 days           
                                                            fluentd-elasticsearch     gcr.io/google_containers/fluentd-elasticsearch:1.5                                                                                                                                          Running   2 days           
kibana-logging-7kcv4                           10.244.3.3                                                                                                                 kubernetes-minion-xf0z/104.197.12.87    kubernetes.io/cluster-service=true,name=kibana-logging          Running   2 days           
                                                            kibana-logging            gcr.io/google_containers/kibana:1.3                                                                                                                                                         Running   2 days           
kube-apiserver-kubernetes-master                                                                                                                                          kubernetes-master/                      <none>                                                          Running   About a minute   
                                                            kube-apiserver            gcr.io/google_containers/kube-apiserver:277fe30e546796064bed598ac243aafa                                                                                                                    Running   About a minute   last termination: exit code 255
kube-controller-manager-kubernetes-master                                                                                                                                 kubernetes-master/                      <none>                                                          Running   About a minute   
                                                            kube-controller-manager   gcr.io/google_containers/kube-controller-manager:b06e6ce789137d0f05e578494badd8f3                                                                                                           Running   About a minute   last termination: exit code 255
kube-dns-i7hay                                 10.244.1.3                                                                                                                 kubernetes-minion-kxcb/104.197.16.133   k8s-app=kube-dns,kubernetes.io/cluster-service=true             Running   2 days           
                                                            skydns                    gcr.io/google_containers/skydns:2015-03-11-001                                                                                                                                              Running   2 days           
                                                            kube2sky                  gcr.io/google_containers/kube2sky:1.4                                                                                                                                                       Running   2 days           
                                                            etcd                      gcr.io/google_containers/etcd:2.0.9                                                                                                                                                         Running   2 days           
kube-scheduler-kubernetes-master                                                                                                                                          kubernetes-master/                      <none>                                                          Running   About a minute   
                                                            kube-scheduler            gcr.io/google_containers/kube-scheduler:847699939480eff0ba7913e04874eb25                                                                                                                    Running   About a minute   last termination: exit code 255
monitoring-heapster-controller-b710j           10.244.1.2                                                                                                                 kubernetes-minion-kxcb/104.197.16.133   kubernetes.io/cluster-service=true,name=heapster                Running   2 days           
                                                            heapster                  gcr.io/google_containers/heapster:v0.11.0                                                                                                                                                   Running   11 seconds       last termination: exit code 1
synthetic-logger-0.25lps-pod                   10.244.3.4                                                                                                                 kubernetes-minion-xf0z/104.197.12.87    name=synth-logging-source                                       Running   2 days           
                                                            synth-lgr                 ubuntu:14.04                                                                                                                                                                                Running   2 days           
synthetic-logger-10lps-pod                     10.244.3.5                                                                                                                 kubernetes-minion-xf0z/104.197.12.87    name=synth-logging-source                                       Running   2 days           
                                                            synth-lgr                 ubuntu:14.04                                                                                                                                                                                Running   2 days           
```

**Comment by jayunit100** [comment]
thanks satnam. i ll try to see why . unfortunately i dont have this sort of cluster so cant really test, but ill look if something pops out. 

indeed i tested it skips, but wasnt able to test that it doesnt skip .

**Comment by satnam6502** [comment]
To prove my point, this is what happens when I run the test against a recent build from HEAD:
```
$ go run hack/e2e.go --v --test --test_args="--ginkgo.focus=.*Cluster.*"
2015/05/11 11:24:30 Running: get status
Project: kubernetes-satnam
Zone: us-central1-b
Client Version: version.Info{Major:"0", Minor:"16+", GitVersion:"v0.16.2-710-g2709bb3987698a-dirty", GitCommit:"2709bb3987698ab63af5d8cc60120fe7cb0eda52", GitTreeState:"dirty"}
Server Version: version.Info{Major:"0", Minor:"16+", GitVersion:"v0.16.2-710-g2709bb3987698a-dirty", GitCommit:"2709bb3987698ab63af5d8cc60120fe7cb0eda52", GitTreeState:"dirty"}
Waiting at most 1200 seconds for all pods to be 'Running'
Found 14 pods with statuses: Running Running Running Running Running Running Running Running Running Running Running Running Running Running
All pods are 'Running'
Project: kubernetes-satnam
Zone: us-central1-b
2015/05/11 11:24:35 Running: Ginkgo tests
Setting up for KUBERNETES_PROVIDER="gce".
Project: kubernetes-satnam
Zone: us-central1-b
Running Suite: Kubernetes e2e Suite run 1 of 1
==============================================
Random Seed: 1431368678 - Will randomize all specs
Will run 1 of 46 specs

SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
Cluster level logging using Elasticsearch 
  should check that logs from pods on all nodes are ingested into Elasticsearch
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:46
[BeforeEach] Cluster level logging using Elasticsearch
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:42
>>> testContext.KubeConfig: /usr/local/google/home/satnam/.kube/config
[It] should check that logs from pods on all nodes are ingested into Elasticsearch
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:46
STEP: Checking the Elasticsearch service exists.
STEP: Checking to make sure the Elasticsearch pods are running
STEP: waiting up to 5m0s for pod elasticsearch-logging-ky96p status to be running
STEP: Checking to make sure we are talking to an Elasticsearch service.
STEP: Checking health of Elasticsearch service.
STEP: Waiting for the pods to succeed.
STEP: waiting up to 5m0s for pod synthlogger-0 status to be success or failure
INFO: No Status.Info for container synth-logger in pod synthlogger-0 yet
INFO: Waiting for pod synthlogger-0 in namespace es-logging-3500 status to be "success or failure" (found "Pending") (39.994035ms)
INFO: No Status.Info for container synth-logger in pod synthlogger-0 yet
INFO: Waiting for pod synthlogger-0 in namespace es-logging-3500 status to be "success or failure" (found "Pending") (5.081380263s)
INFO: No Status.Info for container synth-logger in pod synthlogger-0 yet
INFO: Waiting for pod synthlogger-0 in namespace es-logging-3500 status to be "success or failure" (found "Pending") (10.122069274s)
STEP: Saw pod success
STEP: waiting up to 5m0s for pod synthlogger-1 status to be success or failure
STEP: Saw pod success
STEP: waiting up to 5m0s for pod synthlogger-2 status to be success or failure
STEP: Saw pod success
STEP: waiting up to 5m0s for pod synthlogger-3 status to be success or failure
INFO: No Status.Info for container synth-logger in pod synthlogger-3 yet
INFO: Waiting for pod synthlogger-3 in namespace es-logging-3500 status to be "success or failure" (found "Pending") (39.707774ms)
STEP: Saw pod success
STEP: Checking all the log lines were ingested into Elasticsearch
INFO: After 528.742009ms expecting to find 400 log lines but saw only 200
INFO: After 10.728495598s found all 400 log lines

• [SLOW TEST:62.117 seconds]
Cluster level logging using Elasticsearch
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:47
  should check that logs from pods on all nodes are ingested into Elasticsearch
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/es_cluster_logging.go:46
------------------------------
SSSSSSSSS
Ran 1 of 46 Specs in 62.118 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 45 Skipped I0511 11:25:40.572155   18547 driver.go:96] All tests pass
```

**Comment by satnam6502** [comment]
Would you like me to debug it? I think that might be quicker since I have the clusters at hand? I don't want to steal your thunder!

**Comment by jayunit100** [comment]
sure go for it @satnam6502 :) ! ill try to learn more how these tests work meanwhile

**Comment by satnam6502** [comment]
OK, but I am drowning in other stuff right now so I may not get to this until later this week.

**Comment by jayunit100** [comment]
no prob. well maybe ill steal the thunder rightback , then :)

**Comment by jayunit100** [comment]
I think i see what went wrong.  something along these lines...

When refactoring, i guess i made it so that the testContext is referenced outside of code that is run via ginkgo, which results in a silent failure that causes all the tests not to run.

**Comment by jayunit100** [comment]
After looking deeper, i dont think we can use the `[Skipped]` mechanism in tests at test declarations unless we know apriori that a test shouldnt run (like in density).  If we are skipping for gce/gke, then we *have* to prematurely return from the test - which is unfortunate, because....  im pretty sure  this will make the number of passing tests inflated in cases that some tests are skipped.   

So, ill close this guy :).

**Pull request closed without merging**